modin/core/storage_formats/pandas/aggregations.py

```diff
@@ -0,0 +1,588 @@
+
+        Parameters
+        ----------
+        df : pandas.DataFrame
+            Partition to compute the aggregations for.
+
+        Returns
+        -------
+        Tuple[sums: pandas.DataFrame, sums_of_squares: pandas.DataFrame, count: pandas.DataFrame]
+            A tuple storing DataFrames where each of them holds the result for
+            one of the described aggregations.
+        """
+        sums = df.sum().rename(MODIN_UNNAMED_SERIES_LABEL)
+        sums_of_squares = (df**2).sum().rename(MODIN_UNNAMED_SERIES_LABEL)
+        count = pandas.Series(
+            np.repeat(len(df), len(df.columns)), index=df.columns, copy=False
+        ).rename(MODIN_UNNAMED_SERIES_LABEL)
+        return sums, sums_of_squares, count
+
+    @staticmethod
+    def _compute_nan_aggs(
+        raw_df: np.ndarray, cols: pandas.Index, nan_mask: np.ndarray
+    ) -> Tuple[pandas.DataFrame, pandas.DataFrame, pandas.DataFrame]:
+        """
+        Compute sums, sums of square and the number of observations for a partition assuming there are NaN values in it.
+
+        Parameters
+        ----------
+        raw_df : np.ndarray
+            Raw values of the partition to compute the aggregations for.
+        cols : pandas.Index
+            Columns of the partition.
+        nan_mask : np.ndarray[bool]
+            Boolean mask showing positions of NaN values in the `raw_df`.
+
+        Returns
+        -------
+        Tuple[sums: pandas.DataFrame, sums_of_squares: pandas.DataFrame, count: pandas.DataFrame]
+            A tuple storing DataFrames where each of them holds the result for
+            one of the described aggregations.
+        """
+        # Unfortunately, in case of NaN values we forced to compute multiple sums/square sums/counts
+        # for each column because we have to exclude values at positions of NaN values in each other
+        # column individually.
+        # Imagine we have a dataframe like this:
+        #   col1: 1, 2  , 3  , 4
+        #   col2: 2, NaN, 3  , 4
+        #   col3: 4, 5  , NaN, 7
+        # In this case we would need to compute 2 different sums/square sums/count for 'col1':
+        #   - The first one excluding the values at the NaN possitions of 'col2' (1 + 3 + 4)
+        #   - And the second one excluding the values at the NaN positions of 'col3' (1 + 2 + 4)
+        # and then also do the same for the rest columns. At the end this should form a matrix
+        # of pairwise sums/square sums/counts:
+        #   sums[col1, col2] = sum(col1[i] for i in non_NA_indices_of_col2)
+        #   sums[col2, col1] = sum(col2[i] for i in non_NA_indices_of_col1)
+        #   ...
+        # Note that sums[col1, col2] != sums[col2, col1]
+        sums = {}
+        sums_of_squares = {}
+        count = {}
+
+        # TODO: is it possible to get rid of this for-loop somehow?
+        for i, col in enumerate(cols):
```

Comment:
Maybe something like that:
```python
    col_vals = np.repeat(raw_df, raw_df.shape[0], axis=0)
    res_nan_mask = np.resize(nan_mask, (raw_df.shape[0]**2, raw_df.shape[1]))
    np.putmask(col_vals, res_nan_mask, values=0)
    sums = np.sum(rep_test, axis=1)
    sums = reshape(raw_df.shape[0], sums.shape[0]//raw_df.shape[0], order="F")
    ...
```

Reply:
I've already tried this approach

> col_vals = np.repeat(raw_df, raw_df.shape[0], axis=0)

This line says that you have to replicate the original dataset X times, where X is the number of columns. This maneuver can be really expensive as we have to allocate `X * dataframe_size` memory at once. 

Consider the example above, where we have a dataframe with the shape of (1_552_210, 214) ~ 2.5gb, and then we replicate it 214 times (2.5gb * 214 = 528gb of RAM required). No doubt I got OOM on that.

Reply:
Ok, can https://numpy.org/doc/stable/reference/generated/numpy.apply_along_axis.html#numpy.apply_along_axis help here?